PYME/LMVis/visCore.py

```diff
@@ -426,8 +426,51 @@ def add_layer(self, layer):
         self.glCanvas.refresh()
     
         self.layer_added.send(self)
-
     
+    def generate_test_object_layer(self, obj_type):
+        vx, vy, vz = 0.1, 0.1, 0.1  # um
+        density = 1.0/(vx*vy*vz)
+        if obj_type == 'sphere':
+            # Generate a Gaussian point cloud to approximate a sphere
+            sigma_x, sigma_y, sigma_z = 1.5, 1.5, 1.5  # um
+            sigma = sigma_x*sigma_y*sigma_z
+            n_points = np.round(density*sigma/0.314432).astype(int)  # 0.314432 = 0.68^3
+            x = np.random.randn(n_points) * sigma_x * 1e3  # nm
+            y = np.random.randn(n_points) * sigma_y * 1e3  # nm
+            z = np.random.randn(n_points) * sigma_z * 1e3  # nm
+            t = (np.random.rand(n_points)*n_points).astype(int)  # dummy
+        else:
+            raise ValueError('Unknown object type passed: {}'.format(obj_type))
+
+        from PYME.IO.tabular import DictSource
+        from PYME.IO.MetaDataHandler import CachingMDHandler
+
+        points = DictSource({'x':x,'y':y,'z':z,'t':t})
+        points.mdh = CachingMDHandler({'voxelsize.x': vx, 'voxelsize.y': vy, 'voxelsize.z': vz,
+                                   'voxelsize.units': 'um'})
+
+        print('Creating Pipeline')
+        self.pipeline.OpenFile(ds=points)
+        print('Pipeline Created')
+        
+        #############################
+        #now do all the gui stuff
```

Comment:
Why is this needed - surely we should be able to just hook the standard open and launch code ...

Reply:
I was having trouble hooking the standard open/launch code, which I assume is `visCore.VisGUICore.OpenFile()`. This `now do all the gui stuff` I copied from `visCore.VisGUICore.OpenFile()`. While `pipeline.OpenFile()` allows you to pass a data source as an argument, `visCore.VisGUICore.OpenFile()` is presently limited to files. Perhaps we should add a `ds` flag and/or `**kwargs` and associated handling to `visCore.VisGUICore.OpenFile()`. Or is there another function I missed?